src/data_tools/data_transformer.py

```diff
@@ -0,0 +1,867 @@
+    Applies a gaussian blur to the input image.
+    
+    INPUTS
+    ------
+    image: Input image of dimension WIDTH x HEIGHT.
+    kernel_x and kernel_y: Determine the blurring. 1 is added to both after they are multipled by 2 to ensure the input into the cv2.GaussianBlur function is odd
+        positive integers
+        default values for both are 10.
+    
+    NOTES
+    -----
+    Does not affect labels.
+    '''
+    ret = np.copy(image)
+    return cv2.GaussianBlur(ret, (2*kernel_x + 1, 2*kernel_y + 1), 0)
+
+def colour(image, bf=1, gf=1, rf=1, mode="RGB"):
+    '''
+    Modifies colour channels of input image.
+
+    INPUTS
+    ------
+    image: Input image of dimensions WIDTH x HEIGHT.
+    bf: blue-factor to be multiplied to the blue channel of the image.
+        bf = 0 will remove all blue from the image
+        if bf increases the blue beyond 255, it gets capped at 255
+    gf: green-factor to be multiplied to the green channel of the image.
+        gf = 0 will remove all green from the image
+        if gf increases the green beyond 255, it gets capped at 255  
+    rf: red-factor to be multiplied to the red channel of the image.
+        rf = 0 will remove all red from the image
+        if rf increases the red beyond 255, it gets capped at 255  
+    mode: indicates if you want to modify the RGB channels of your image.
+        default modifies RGB channels
+        otherwise, applies grayscale to the image
+
+    NOTES
+    -----
+    Does not affect labels.
+
+    '''
+    if (mode != "RGB"):
+        return cv2.cvtColor(image, cv2.COLOR_BGR2GRAY)
+    else:
+        ret = np.copy(image)
+        if (bf <= 1):
+            ret[:,:,0] = (ret[:,:,0]*bf).astype(int)
+        else:
+            blue = np.copy(image)
+            blue[:,:,0] = 255
+            ret[:,:,0] = ((ret[:,:,0] + 2*blue[:,:,0])/3).astype(int)
+        if (gf <= 1):
+            ret[:,:,1] = (ret[:,:,1]*gf).astype(int)
+        else: 
+            green = np.copy(image)
+            green[:,:,1] = 255
+            ret[:,:,1] = ((ret[:,:,1] + 2*green[:,:,1])/3).astype(int)
+        if (rf <= 1):
+            ret[:,:,2] = (ret[:,:,2]*rf).astype(int)
+        else:
+            red = np.copy(image)
+            red[:,:,2] = 255
+            ret[:,:,2] = ((ret[:,:,2] + 2*red[:,:,2])/3).astype(int)
+        
+        return ret
+
+def pixel_swap(image, seed, swaps):
+    '''
+    Randomly swaps pixels of an image to create noise.
+
+    INPUTS
+    ------
+    image: Input image of dimensions WIDTH x HEIGHT.
+    seed: Allows you to repreduce results. If called with the same seed twice, the output will be the same.
+        any integer
+    swaps: determines the number of times pixels are swapped.
+        any integer
+    
+    NOTES
+    -----
+    WARNING VERY SLOW
+    Must be applied to labels as well.
+    '''
+    height, width, channels = image.shape
+    ret = np.copy(image)
+    for num in range(0, swaps):
+        random.seed(seed + num)
```

Comment:
Oh, you only need to set the random seed once. Probably better to set it outside of the for-loop (faster that way). It should be random enough with just setting the seed once :)